### ISSUE TITLE
Health-Setter does not work after your Player respawned 1x

### ISSUE THREAD

**Issue by cRiZZlyy-CR** [altmp/altv-issues#1775]
### Description of the problem

Hello,

I was testing some Code & i figured out, that when you set the Health on 0 while Changing Example Weapons like me: 

That after you respawned 1x the Health Setter does not work & just triggering my Code without a Respawn.

### Reproduction steps

1. Set the Player Health to 0 & call with the Event "OnPlayerDead" -> If he dies, Respawn him & log the Respawn with a Position

You will see, If you died 1x time & reproduce your Code again, that will show you, you are not getting killed again & respawned

### Expected behaviour

You should respawn all time after the Health Setter is at 0

### Additional context

My Code:

**OnWeaponChange:**

    internal class WeaponChange : IScript
    {
        [AsyncScriptEvent(ScriptEventType.PlayerWeaponChange)]
        public Task OnPlayerWeaponChange(IPlayer CPlayer, uint oldWeapon, uint newWeapon)
        {
            Alt.Log($"{CPlayer.Name} changed {oldWeapon} {newWeapon}!");

            CPlayer.Health = 0;
            return Task.CompletedTask;
        }
    }
}

**PlayerDied:**

    internal class PlayerDead : IScript
    {
        [AsyncScriptEvent(ScriptEventType.PlayerDead)]
        public Task OnPlayerDeadAsync(IPlayer CPlayer, IEntity killer, uint weapon)
        {
            Alt.Log($"{CPlayer.Name} died!");
            CPlayer.Spawn(new Position(-425, 1123, 325));
            Alt.Log($"{CPlayer.Name} Respawned!");
            return Task.CompletedTask;
        }
    }
}

![Screenshot 2023-03-09 192120](https://user-images.githubusercontent.com/91555896/224119669-1fb455bf-74a9-4953-8cd0-8096e19a35d4.png)

### Operating system

Windows 11 64

### Version

Release

### Reproduction tested

- [X] I confirm that I have made sure that this issue is also present on the newest dev version

**Comment by litolax** [comment]
Confirmed on the latest dev 14.0-dev57 
![image](https://user-images.githubusercontent.com/60060712/224112747-04f853bf-7a42-4971-8d17-97e4a96ddcea.png)

**Comment by xxshady** [comment]
Workaround:
```js
alt.on("playerWeaponChange", (player, old, weapon) => {
  player.health = (player.health === 0) ? 1 : 0
})
```

**Comment by BluePeer** [comment]
that's just a bad implementation because you don't consider what's happening there.

until the end of the weapon change event, the character is still equipped with the non-default weapon hand

since you set the HP to 0 in this event
do you trigger again without the weapon being replaced by the hand (resync) (which only happens at the end of the event) that the charstate is retransmitted (contains the old weapon with the new state dead)
that you send to the client first
-"You are dead" transfer of the char status change from life to dead with old weapon because it is still in the event and has not yet changed.
- "Weapon change event through" (char status weapon from old to new)

that means your client is up to date
-ClientChar(HP: 100, WeaponId: 1)
you change the weapon to id:2
-ClientChar(HP: 100, WeaponId: 2) -> send weaponchange event 1->2
server sets to dead in event
-ClientChar(HP: 0, WeaponId: 1) -> send weapon change event 2->1
server sets weapon to 2 after change event
-ClientChar(HP: 0, WeaponId: 2) -> send weapon change event 1->2
loooooop

**Comment by FabianTerhorst** [comment]
Fixed in dev/15.0-dev333
